src/components/contact-datalist-item/ContactDatalistItem.js

```diff
@@ -7,31 +7,86 @@ import DatalistItem from '../datalist-item';
 import './ContactDatalistItem.scss';
 
 type Props = {
-    getContactAvatarUrl?: (contact: { id: string, [key: string]: any }) => string,
+    getContactAvatarUrl?: (contact: { id: string, [key: string]: any }) => string | Promise<?string>,
     id?: string,
     isExternal?: boolean,
     name: ?string,
     showAvatar?: boolean,
     subtitle?: React.Node,
 };
 
-const ContactDatalistItem = ({ getContactAvatarUrl, id, isExternal, name, showAvatar, subtitle, ...rest }: Props) => (
-    <DatalistItem className="contact-data-list-item" {...rest}>
-        {showAvatar && (
-            <Avatar
-                className="contact-avatar"
-                id={id}
-                name={name}
-                isExternal={isExternal}
-                shouldShowExternal
-                avatarUrl={getContactAvatarUrl && id ? getContactAvatarUrl({ id }) : undefined}
-            />
-        )}
-        <div className="contact-name-container">
-            <div className="contact-text contact-name">{name}</div>
-            {subtitle && <div className="contact-text contact-sub-name">{subtitle}</div>}
-        </div>
-    </DatalistItem>
-);
+type State = {
+    avatarUrl: ?string,
+};
+
+class ContactDatalistItem extends React.PureComponent<Props, State> {
+    state = {
+        avatarUrl: undefined,
+    };
+
+    isMounted: boolean = false;
+
+    /**
+     * Success handler for getting avatar url
+     *
+     * @param {string} avatarUrl the user avatar url
+     */
+    getAvatarUrlHandler = (avatarUrl: ?string) => {
+        if (this.isMounted) {
+            this.setState({
+                avatarUrl,
+            });
+        }
+    };
+
+    /**
+     * Gets the avatar URL for the user from the getAvatarUrl prop
+     *
+     * @return {string} the avatar url string
```

Comment:
This isn't true. It will not return anything; instead it will trigger a promise that actually returns the string.

Reply:
my mistake forgot to change that